pkg/storaged/btrfs/volume.jsx

```diff
@@ -0,0 +1,190 @@
+import { dialog_open, TextInput } from "../dialog.jsx";
+import { make_btrfs_subvolume_page } from "./subvolume.jsx";
+import { mounting_dialog } from "../filesystem/mounting-dialog.jsx";
+
+const _ = cockpit.gettext;
+
+function volume_mount(volume) {
+    const block = client.blocks[volume.path];
+    mounting_dialog(client, block, "mount", null);
+}
+
+/*
+ * Udisks is a disk/block library so it manages that, btrfs turns this a bit
+ * around and has one "volume" which can have multiple blocks by a unique uuid.
+ *
+ * Cockpit shows Btrfs as following:
+ *
+ * -> btrfs subvolume
+ *    -> btrfs volume
+ *        -> btrfs device
+ *            -> block device
+ */
+export function make_btrfs_volume_card(parent, uuid) {
+    const block_devices = client.uuids_btrfs_blocks[uuid];
+    const block_btrfs = client.blocks_fsys_btrfs[block_devices[0].path];
+    const volume = client.uuids_btrfs_volume[uuid];
+    const use = btrfs_usage(client, volume);
+
+    const btrfs_disks_card = new_card({
+        title: _("btrfs devices"),
+        next: null,
+        component: BtrfsDevicesCard,
+        props: { uuid },
+    });
+
+    const name = block_btrfs.data.label || uuid;
+    const btrfs_volume_card = new_card({
+        title: _("btrfs volume"),
+        next: btrfs_disks_card,
+        page_location: ["btrfs-volume", uuid],
+        page_name: name,
+        page_icon: VolumeIcon,
+        page_size: use[1],
+        component: BtrfsVolumePage,
+        props: { block_devices, uuid, use },
+    });
+
+    const actions = [];
+    const is_mounted = btrfs_is_volume_mounted(client, block_devices);
+    if (!is_mounted) {
+        actions.push({
+            title: _("Mount"),
+            action: () => volume_mount(volume),
+        });
+    }
+
+    const subvolumes_card = new_card({
+        title: _("btrfs subvolumes"),
+        next: btrfs_volume_card,
+        component: BtrfsSubVolumesCard,
+        props: { volume },
+        // TODO: create subvolume
+        actions,
+    });
+
+    const subvolumes_page = new_page(parent, subvolumes_card);
+    make_btrfs_subvolume_pages(subvolumes_page, volume);
+}
+
+const BtrfsVolumePage = ({ card, block_devices, uuid, use }) => {
+    const block_btrfs = client.blocks_fsys_btrfs[block_devices[0].path];
+    const label = block_btrfs.data.label || "-";
+
+    // Changing the label is only supported when the device is not mounted
+    // otherwise we will get btrfs filesystem error ERROR: device /dev/vda5 is
+    // mounted, use mount point This is a libblockdev/udisks limitation as it
+    // only passes the device and not the mountpoint when the device is mounted.
+    // https://github.com/storaged-project/libblockdev/issues/966
+    const is_mounted = btrfs_is_volume_mounted(client, block_devices);
+
+    function rename_dialog() {
+        dialog_open({
+            Title: _("Change label"),
+            Fields: [
+                TextInput("name", _("Name"),
```

Comment:
These 5 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19690-20231205-110913-7b3be2c9-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)